src/lapi.cc

```diff
@@ -126,6 +128,59 @@ namespace blt {
             return 1;
         }
 
+        int
+        movedir(lua_state* state)
+        {
+            size_t fromLen, toLen;
+            const char* from = lua_tolstring(state, 1, &fromLen);
+            const char* to   = lua_tolstring(state, 2, &toLen);
+
+            if(lua_gettop(state) != 2)
+                luaL_error(state, "file.MoveDirectory(from, to) takes two arguments, not %d", lua_gettop(state));
+
+            if(strlen(from) != fromLen)
+                luaL_error(state, "file.MoveDirectory(from, to): argument 'from' cannot contain null characters!");
+
+            if(strlen(to) != toLen)
+                luaL_error(state, "file.MoveDirectory(from, to): argument 'to' cannot contain null characters!");
+
+            bool result = (rename(from, to) == 0);
+            lua_pushboolean(state, result);
+            if(!result) {
```

Comment:
See my PR comment about brackets

Reply:
Yeah, I realized near the start that they should go on a seperate line, and changed them, but I must have missed a few. I'll fix that now too.

Reply:
ok

Reply:
How do you format catch(...) blocks?

Reply:
just like if/else